server/src/test/java/org/candlepin/dto/AbstractDTOTest.java

```diff
@@ -97,6 +102,16 @@ public T getDTOInstance() {
         }
     }
 
+    public void addCollectionMethod(Class<T> dtoClass, String field, String prefix,
```

Comment:
I'm not opposed to generalizing all of the collection-based tests, but I'd like to see this cleaned up a bit and not as hacky as it is. Right now the field name is somewhat misused to fetch a value using one of two values, and the method lookup is dependent on a very specific pluralization, which may not always hold true. I.e. content is already plural, some pluralization might require an -es, -ies or just an -s suffix.

More work needs to be done here to make this work cleanly, and is a large part of why it wasn't done initially.

Reply:
I hear you. noting this as a topic of discussion.

Reply:
One way of approaching this problem is changing the detection to use the prefix (add/remove/get) with a similar noun (i.e. Products or Content), and then add them to a separate list of things to test (as you're doing)